packages/firestore/src/util/async_queue.ts

```diff
@@ -23,6 +23,18 @@ import { Code, FirestoreError } from './error';
 // tslint:disable-next-line:no-any Accept any return type from setTimeout().
 type TimerHandle = any;
 
+/**
+ * Wellknown "timer" IDs used when scheduling delayed operations on the
+ * AsyncQueue. These IDs can then be used from tests to check for the presence
+ * of operations or to run them early.
+ */
+export enum TimerId {
```

Comment:
I'm not sure I am a big fan of this Enum. Can we use well-defined strings kind of like we for the action strings in `Persistence.runTransaction`?  It seems strange that we would have to edit this type everytime we add a new f task.

Reply:
I started out with a string union (`type TimerId = 'listen_stream_idle'|...`), but it was feeling awkward to have long string constants that had to match between our tests and implementation code.  And since iOS / Java don't have string unions I assume we'd want to use an enum there? So I just went with an enum.

I don't follow "It seems strange that we would have to edit this type everytime we add a new task."  Wouldn't the same be true if we used a string union? Or did you have something else in mind?

FWIW, I think the runTransaction strings are just used for logging, so they don't have the same constraint of being used from multiple places.